File: python/67b_1445A_ArrayRearrangment.py
def solve(a, b, n, x):
    chk = [0] * 1001
    for i in a:
       chk[i] += 1
    for i in range(1, a[n-1]+1):
        chk[i] += chk[i-1]
    #print(chk)
    for i in range(n-1, -1, -1):
        k = x - b[i]
        if chk[k] < (n-i):
            return "NO"
    return "YES" 
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n, x = map(int, input().split())
        a = list(map(int, input().split()))
        b = list(map(int, input().split()))
        c = input()
        ans.append(solve(a, b, n, x))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/67b_1445A_ArrayRearrangment.cpp
#include<bits/stdc++.h>

using namespace std;

string solve(vector<int> a, vector<int> b, int n, int x){
    int k;
    vector<int> chk(1001, 0);
    for(auto i : a){
        chk[i]++;
    }
    for(int i = 1; i < 1000; i++){
        chk[i] += chk[i-1];
    }
    for(int i = 0; i < n; i++){
        k = x - b[i];
        if(chk[k] < (n-i)){
            return "NO";
        }
    }
    return "YES";
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n, x, temp;
    vector<string> ans;
    cin>>t;

    for(int i = 0; i < t; i++){
        cin>>n>>x;
        vector<int> a, b;
        for(int j = 0; j < n; j++){
            cin>>temp;
            a.push_back(temp);
        }
        for(int j = 0; j < n; j++){
            cin>>temp;
            b.push_back(temp);
        }
        ans.push_back(solve(a, b, n, x));
    }
    for(auto i : ans){
        cout<<i<<"\n";
    }
    return 0;
}